src/modules/ModuleEIP712.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.25;
+
+// A base for modules that use EIP-712 structured data signing.
+//
+// Unlike other EIP712 libraries, this mixin uses the salt field to hold the account address.
+//
+// It also includes the name, but not a version, in the EIP-712 domain, as modules are intended to be deployed as
+// immutable singletons, thus a different version would have a different module address in the salt.
```

Comment:
Is this a safe assumption to make?

I'm not sure-- could we use the remaining 12 bytes in the salt to store a version, or is that overkill?

Reply:
I think it's fair for this mixin to be incompatible with upgradeable modules - that's a really niche use case, and often incompatible with 4337. This did point out that I need to update the comment though, because we're not using the name field.

I think a point of comparison could be the EIP-712 format for Safe transactions ([here](https://github.com/safe-global/safe-smart-account/blob/v1.4.1/contracts/Safe.sol#L49-L57)) - they only use a verifyingContract and chainId in the domain. If a module needs a name or versioning for differentiation, it's still possible to encode in the struct itself, or to not use this mixin and use a different domain separator.

Reply:
Makes sense, but thinking back-- do we actually prevent upgradeable modules this way? 

I don't think it's prevented-- you're just not differentiating signatures with one or the other implementation. But, even then, you could always implement a custom domain separator in your upgraded impl, yeah?

Reply:
> do we actually prevent upgradeable modules this way?

Yeah this doesn't prevent it, it's just (probably) wrong to use this with upgradeable modules.